http/middleware.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 Democratized Data Foundation
```

Comment:
praise: I really like how you have solved some of the complexities with Lens/Collections/txns/etc in this file :)

todo: I do think the documentation within this file needs to be expanded though, particularly on the stuff that is accessed from within the handlers (i.e. the `fooContextKey` variables).  It is not obvious when consuming them whether stuff like txns are respected.

Reply:
Added more documentation clarifying what each context key will be set to.

Reply:
Looks good - thanks Keenan